List modules in README. Give each module its own README.

leifwickland opened the pull request:
Fixes #120.

leifwickland committed:
List modules in README. Give each module its own README.

Fixes #120.

melrief reviewed (approved):
LGTM but before I merge I was thinking that maybe Integration alone is not enough as title. What about Integration with other libraries? It's much easier to understand what it is about, I think. What do you think about it?

jcazevedo commented on README.md:
```diff
@@ -130,13 +131,42 @@ loadConfig[MyClass](conf)
+
+### Internal Modules 
+
+The core of PureConfig eschews unnessary dependencies. Separate modules exist provide support for types which are not part of the standard Scala and Java libraries, 
```
I think there's a `to` missing here: `Separate modules exist to provide (...)`.

jcazevedo commented on README.md:
```diff
@@ -130,13 +131,42 @@ loadConfig[MyClass](conf)
+3. Add a `modules/nextbestmod/build.sbt` defining the module's name and special dependencies.
+4. Implement converters in a `package object` in `modules/nextbestmod/src/main/scala/pureconfig/module/nextbestmod/package.scala`.
+5. Test the converters in `modules/nextbestmod/src/test/scala/pureconfig/module/nextbestmod/NextBestModuleSuite.scala`.
+6. Optionally explain a little bit about how it works in `modules/nextbestmod/README.md`
```
This last point is missing a period at the end.

jcazevedo commented on README.md:
```diff
@@ -130,13 +131,42 @@ loadConfig[MyClass](conf)
 [available issues](https://github.com/melrief/pureconfig/issues) and to talk with
 us on the [pureconfig gitter channel](https://gitter.im/melrief/pureconfig?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge).
 
+If you'd like to add support for types which are not part the standard Java or Scala libraries, please consider submitting a pull request to create a [module](#internal-modules). [Pull Request #108](https://github.com/melrief/pureconfig/pull/108/files) created a very simply module. It should provide a good template for the pieces you'll need to add. 
```
I think you mean `very simple module` instead of `very simply module`.

jcazevedo commented on README.md:
```diff
@@ -130,13 +131,42 @@ loadConfig[MyClass](conf)
+2. Create a new  `modules/nextbestmod/` subdirectory.
+3. Add a `modules/nextbestmod/build.sbt` defining the module's name and special dependencies.
+4. Implement converters in a `package object` in `modules/nextbestmod/src/main/scala/pureconfig/module/nextbestmod/package.scala`.
+5. Test the converters in `modules/nextbestmod/src/test/scala/pureconfig/module/nextbestmod/NextBestModuleSuite.scala`.
```
Do we want to explicitly tell where converters and tests should go?

jcazevedo commented on modules/joda/README.md:
```diff
@@ -0,0 +1,45 @@
+
+## Why
+
+Create configurable converters to read [Joda Time](http://www.joda.org/joda-time/) types from from configuration.
```
`from` is repeated twice.

jcazevedo reviewed (changes requested):
LGTM overall. I left some comments, mostly related to typos.

I agree with @melrief that "Integration with other libraries" or "Integrating with other libraries" is a more explicit title for the section.

leifwickland committed:
Change section header. Fix typos and trailing whitespace

leifwickland commented:
@melrief I updated the section header as you suggested. I didn't add anything to "Contributing" about how publishing because I wasn't entirely sure what you wanted to say about that. Do you have a sentence or two you want me to include?

@ruippeixotog Thanks for spotting my typos. I think I've fixed those and addressed your concern about excessively prescriptive language about where converters are implemented and tested.

ruippeixotog commented on README.md:
```diff
@@ -130,13 +131,42 @@ loadConfig[MyClass](conf)
+
+The core of PureConfig eschews unnessary dependencies. Separate modules exist to support types which are not part of the standard Scala and Java libraries.
+
+- [`pureconfig-enum`](modules/enum/) provides converters for enums generated by [julienrf's enum library](https://github.com/julienrf/enum).
```
Minor (OCD-level, in fact) nitpick: in other places in the README we end each list item with a `;` except the last one, which ends in `.`. Can you do that here too?

leifwickland replied:
@ruippeixotog 
(Pedantic-level response)
Technically I think it'd be incorrect to use a semicolon in this case because each item of the list is a complete sentence. Although Henry David Thoreau might string together semi-colon delimited complete thoughts as a paragraph, in modern usage it'd be weird.
I believe that's in contrast to our other bulleted list in the README which is a series of predicate nominatives following a colon.
That said, if you're passionate about it, I'm happy to change it.

ruippeixotog replied:
Makes sense. Disregard what I said :)

ruippeixotog commented on README.md:
```diff
@@ -130,13 +131,42 @@ loadConfig[MyClass](conf)
+
+The steps to create a new module, called _`nexttopmod`_, are:
+
+1. Define a new project in the root `build.sbt`. There are other examples near the top of the file.
```
Same thing here.

leifwickland replied:
Same pedantic response. Still happy to change it if you prefer.

ruippeixotog reviewed (commented):
LGTM, nice job :) I like how each module ended up with its own README page and the main README links to all the other ones. Just take a look at my two minor comments.

melrief commented:
Nice job @leifwickland !
